File: docs/versioned_docs/version-1.10.0/API-Reference/python-examples/api-flows/create-flow.py
import os

import requests

url = f"{os.getenv('LANGFLOW_URL', '')}/api/v1/flows/"

headers = {
    "accept": "application/json",
    "Content-Type": "application/json",
    "x-api-key": f"{os.getenv('LANGFLOW_API_KEY', '')}",
}

payload = {
    "name": "string2",
    "description": "string",
    "icon": "string",
    "icon_bg_color": "#FF0000",
    "gradient": "string",
    "data": {},
    "is_component": False,
    "updated_at": "2024-12-30T15:48:01.519Z",
    "webhook": False,
    "endpoint_name": "string",
    "tags": ["string"],
}

response = requests.request("POST", url, headers=headers, json=payload)
response.raise_for_status()

print(response.text)

File: docs/versioned_docs/version-1.12.0/API-Reference/javascript-examples/api-flows/create-flow.js
const url = `${process.env.LANGFLOW_URL ?? ""}/api/v1/flows/`;

const options = {
  method: 'POST',
  headers: {
    "accept": `application/json`,
    "Content-Type": `application/json`,
    "x-api-key": `${process.env.LANGFLOW_API_KEY ?? ""}`,
  },
  body: JSON.stringify({
  "name": "string2",
  "description": "string",
  "icon": "string",
  "icon_bg_color": "#FF0000",
  "gradient": "string",
  "data": {},
  "is_component": false,
  "updated_at": "2024-12-30T15:48:01.519Z",
  "webhook": false,
  "endpoint_name": "string",
  "tags": [
    "string"
  ]
}),
};

fetch(url, options)
  .then(async (response) => {
    if (!response.ok) {
      throw new Error(`HTTP ${response.status}`);
    }
    const text = await response.text();
    console.log(text);
  })
  .catch((error) => console.error(error));
